server/controllers/peopleController.ts

```diff
@@ -1,17 +1,43 @@
 import type { Request, RequestHandler, Response } from 'express'
 import paths from '../paths/apply'
+import { errorMessage, errorSummary } from '../utils/validation'
+import PersonService from '../services/personService'
 
 export default class PeopleController {
+  constructor(private readonly personService: PersonService) {}
+
   find(): RequestHandler {
     return async (req: Request, res: Response) => {
       const { crn } = req.body
 
       if (crn) {
-        // TODO: error handling if invalid/not found CRN
-        res.redirect(paths.applications.show({ crn }))
+        try {
+          await this.personService.findByCrn(req.user.token, crn)
+
+          res.redirect(paths.applications.show({ crn }))
+        } catch (err) {
+          if (err.status === 404) {
+            this.addErrorMessagesToFlash(req, `No person with a CRN of '${crn}' was found`)
+          } else if (err.status === 403) {
+            this.addErrorMessagesToFlash(req, 'You do not have permission to access this CRN')
+          } else {
+            throw err
+          }
+
+          res.redirect(req.headers.referer || '')
```

Comment:
this is my bad, I left in the empty string earlier because typescript was complaining, we can remove it now (and the test that checks for it in `peopleController.test.ts`
```suggestion
          res.redirect(req.headers.referer)
```

Reply:
ah nice one, thanks